src/Control/Effect/Error.hs

```diff
@@ -76,6 +77,10 @@ instance Monad m => Monad (ErrorC e m) where
   ErrorC a >>= f = ErrorC (a >>= either (pure . Left) (runError . f))
   {-# INLINE (>>=) #-}
 
+instance MonadFix m => MonadFix (ErrorC e m) where
+  mfix f = ErrorC (mfix (runError . either (error "mfix (ErrorC): function returned failure") f))
```

Comment:
Like `ExceptT`, this is safe to use provided that you ensure that throwing an error means you won’t inspect the argument.